src/components/ParserComponent.tsx

```diff
@@ -121,6 +114,40 @@ export default class ParserComponent extends React.Component<ParserInterface,
       );
     }
 
+    /**
+     * Purpose: create a mock file event of the actual file for
+     * parse when it is recalled everytime the date format is changed
+     * @preconditions: a file with valid file data and
+     * a valid file name and a valid file type
+     * @return {any}: a mock file event of the actual file
+     */
+    createNewMockFile(): any {
```

Comment:
Why do we need a mock file? Why can't it just be undefined if it's not set?

Reply:
we need mock file event to put in parse as it needs a file event argument